torch/_dynamo/convert_frame.py

```diff
@@ -758,6 +765,8 @@ def log_bytecode(
                 if attempt > 100:
                     unimplemented("100+ RestartAnalysis() calls")
             except exc.SkipFrame as e:
+                if not isinstance(e, exc.TensorifyScalarRestartAnalysis):
+                    TensorifyState.clear()
```

Comment:
I'm not sure how to refactor this but having to manually clear the states here seems error prone (and I also remember stubbing my toe on this with compiler collectives, see https://github.com/pytorch/pytorch/pull/131983